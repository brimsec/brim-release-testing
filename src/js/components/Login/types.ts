export type FormData = {
  host: string
  username: string
  password: string
  save: boolean
}
