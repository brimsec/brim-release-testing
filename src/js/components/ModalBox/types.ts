import {ModalName} from "../../state/Modal/types"
import {RefAttributes} from "react"

export type ModalButton = {label: string; click: Function}
export type ModalButtonTemplate = void | string | ModalButton[]
export type ModalBoxProps = {
  children: any
  title: string
  name: ModalName
  className?: string
  buttons: ModalButtonTemplate
  onClose?: Function
}

export type ModalContentsProps = {
  children: any
  title: string
  className?: string
  buttons: ModalButtonTemplate
  onClose?: Function
} & RefAttributes<HTMLDivElement>
