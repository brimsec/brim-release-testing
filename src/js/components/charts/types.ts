import {Interval} from "../../types"

export type DateSpan = [Date, Date]

export type Margins = {
  top: number
  left: number
  right: number
  bottom: number
}

type Redraw = (arg0: any) => void
type PenFunc = (arg0: any, arg1: Redraw) => void

export type Pen = {
  draw: PenFunc
  mount: (arg0: Element) => void
}

type HistogramState = {
  selection?: DateSpan | null | undefined
  isFetching?: boolean
  isEmpty?: boolean
  isDragging: boolean
}

export type HistogramData = {
  points: {
    ts: Date
    paths: {
      [key: string]: number
    }
    count: number
  }[]
  keys: string[]
  interval: Interval
  span: DateSpan
}

export type HistogramChart = {
  height: number
  width: number
  margins: Margins
  data: HistogramData
  state: HistogramState
  yScale: d3.ScaleLinear<any, any>
  xScale: d3.ScaleTime<any, any>
  pens: Pen[]
}

export type Chart = HistogramChart
