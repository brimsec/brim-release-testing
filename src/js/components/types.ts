export type PassProps = {
  children?: any
  className?: string
}
