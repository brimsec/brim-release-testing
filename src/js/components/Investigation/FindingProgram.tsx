import {isEmpty} from "lodash"
import React from "react"

import {SearchRecord} from "../../types"

type Props = {search: SearchRecord}

export default function FindingProgram({search}: Props) {
  if (isEmpty(search.pins) && isEmpty(search.program))
    return (
      <div className="program">
        <p>*</p>
      </div>
    )

  return (
    <div className="program">
      {search.pins.map((text, i) => (
        <p className="pin" key={i}>
          {text}
        </p>
      ))}
      <p>{search.program}</p>
    </div>
  )
}
