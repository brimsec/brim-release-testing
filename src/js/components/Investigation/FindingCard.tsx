import {includes} from "lodash"
import {useDispatch, useSelector} from "react-redux"
import React from "react"
import ReactTooltip from "react-tooltip"
import classNames from "classnames"
import get from "lodash/get"
import styled from "styled-components"

import {Finding} from "../../state/Investigation/types"
import {globalDispatch} from "../../state/GlobalContext"
import {submitSearch} from "../../flows/submitSearch/mod"
import Current from "../../state/Current"
import FindingProgram from "./FindingProgram"
import Investigation from "../../state/Investigation"
import MagnifyingGlass from "../../icons/MagnifyingGlass"
import Search from "../../state/Search"
import Spaces from "../../state/Spaces/selectors"
import Warning from "../icons/warning-sm.svg"
import usePopupMenu from "../hooks/usePopupMenu"

const StyledMagnifyingGlass = styled(MagnifyingGlass)`
    fill: ${(props) => props.theme.colors.lead};
    min-width: 13px;
    width: 13px;
    min-height: 13px;
    height: 13px;
  }
`

type Props = {finding: Finding}

export default React.memo<Props>(function FindingCard({finding}: Props) {
  const dispatch = useDispatch()
  const clusterId = useSelector(Current.getConnectionId)
  const spaceIds = useSelector(Spaces.ids(clusterId))
  const findingSpaceName = get(finding, ["search", "spaceName"], "")

  function onClick() {
    dispatch(Search.restore(finding.search))
    dispatch(submitSearch({history: true, investigation: false}))
  }

  function renderWarning() {
    const findingSpaceId = get(finding, ["search", "spaceId"], "")
    const tip = "This space no longer exists"

    if (includes(spaceIds, findingSpaceId)) return null

    return (
      <div
        className="warning-body"
        data-tip={tip}
        data-effect="solid"
        data-place="right"
      >
        <Warning />
        <ReactTooltip />
      </div>
    )
  }

  const menu = usePopupMenu([
    {
      label: "Delete",
      click: () => globalDispatch(Investigation.deleteFindingByTs(finding.ts))
    },
    {type: "separator"},
    {
      label: "Delete All",
      click: () => globalDispatch(Investigation.clearInvestigation())
    }
  ])

  return (
    <div
      className={classNames("finding-card-wrapper")}
      onClick={onClick}
      onContextMenu={() => menu.open()}
      title={findingSpaceName}
    >
      <div className="finding-card">
        <StyledMagnifyingGlass />
        <FindingProgram search={finding.search} />
        {renderWarning()}
      </div>
    </div>
  )
})
