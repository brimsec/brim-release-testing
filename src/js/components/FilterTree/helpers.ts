import {isEqual} from "lodash"

import {InvestigationState} from "../../state/Investigation/types"
import Tree from "../../models/Tree"

export function createInvestigationTree(investigation: InvestigationState) {
  const tree = new Tree({data: "ROOT", children: []})

  for (const finding of investigation) {
    let node = tree.getRoot()

    eachFilter(finding.search, (filter) => {
      if (!node) return
      const nextNode = node.children.find((child) =>
        isEqual(child.data.filter, filter)
      )
      if (nextNode) {
        node = nextNode
      } else {
        node = node.addChild({filter, finding})
      }
    })
  }

  return tree
}

function eachFilter(searchBar, callback) {
  const {pins, program} = searchBar
  const filters = [...pins]

  if (!/^\s*$/.test(program)) filters.push(program)

  return filters.forEach(callback)
}
