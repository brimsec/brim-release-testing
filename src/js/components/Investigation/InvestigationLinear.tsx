import React from "react"

import {getKey} from "../../lib/finding"
import {useFindings} from "./useFindings"
import AnimateChildren from "../AnimateChildren"
import FindingCard from "./FindingCard"
import EmptySection from "../common/EmptySection"
import BookIcon from "../../icons/BookSvgIcon"

export default React.memo<{}>(function InvestigationLinear() {
  const findings = useFindings()

  const cards = []

  findings.forEach((f) => {
    cards.push(<FindingCard key={getKey(f)} finding={f} />)
  })

  if (cards.length === 0)
    return (
      <EmptySection
        icon={<BookIcon />}
        message="As you search through your data, your history will appear here."
      />
    )

  return (
    <AnimateChildren className="investigation-linear">{cards}</AnimateChildren>
  )
})
