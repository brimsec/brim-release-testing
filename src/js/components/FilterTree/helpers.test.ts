import {createInvestigationTree} from "./helpers"
import brim from "../../brim"
import {Finding} from "src/js/state/Investigation/types"

function search(pins, program): Finding {
  return {
    ts: brim.time(new Date(0)).toTs(),
    search: {
      pins,
      program,
      spaceName: "default",
      spaceId: "defaultId",
      spanArgs: brim.dateTuple([new Date(1), new Date(2)]).toSpan(),
      target: "events"
    }
  }
}

test("Remove all pins and make a new current", () => {
  const tree = createInvestigationTree([
    search(["a", "b"], "c"),
    search(["a", "b"], "d"),
    search(["a", "b"], "d"),
    search(["a"], "d"),
    search([], "e")
  ])

  expect(tree.toJSON()).toMatchSnapshot()
})
