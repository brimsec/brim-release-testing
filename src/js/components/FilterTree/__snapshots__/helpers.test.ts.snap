// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Remove all pins and make a new current 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "children": Array [],
              "data": Object {
                "filter": "c",
                "finding": Object {
                  "search": Object {
                    "pins": Array [
                      "a",
                      "b",
                    ],
                    "program": "c",
                    "spaceId": "defaultId",
                    "spaceName": "default",
                    "spanArgs": Array [
                      Object {
                        "ns": 1000000,
                        "sec": 0,
                      },
                      Object {
                        "ns": 2000000,
                        "sec": 0,
                      },
                    ],
                    "target": "events",
                  },
                  "ts": Object {
                    "ns": 0,
                    "sec": 0,
                  },
                },
              },
            },
            Object {
              "children": Array [],
              "data": Object {
                "filter": "d",
                "finding": Object {
                  "search": Object {
                    "pins": Array [
                      "a",
                      "b",
                    ],
                    "program": "d",
                    "spaceId": "defaultId",
                    "spaceName": "default",
                    "spanArgs": Array [
                      Object {
                        "ns": 1000000,
                        "sec": 0,
                      },
                      Object {
                        "ns": 2000000,
                        "sec": 0,
                      },
                    ],
                    "target": "events",
                  },
                  "ts": Object {
                    "ns": 0,
                    "sec": 0,
                  },
                },
              },
            },
          ],
          "data": Object {
            "filter": "b",
            "finding": Object {
              "search": Object {
                "pins": Array [
                  "a",
                  "b",
                ],
                "program": "c",
                "spaceId": "defaultId",
                "spaceName": "default",
                "spanArgs": Array [
                  Object {
                    "ns": 1000000,
                    "sec": 0,
                  },
                  Object {
                    "ns": 2000000,
                    "sec": 0,
                  },
                ],
                "target": "events",
              },
              "ts": Object {
                "ns": 0,
                "sec": 0,
              },
            },
          },
        },
        Object {
          "children": Array [],
          "data": Object {
            "filter": "d",
            "finding": Object {
              "search": Object {
                "pins": Array [
                  "a",
                ],
                "program": "d",
                "spaceId": "defaultId",
                "spaceName": "default",
                "spanArgs": Array [
                  Object {
                    "ns": 1000000,
                    "sec": 0,
                  },
                  Object {
                    "ns": 2000000,
                    "sec": 0,
                  },
                ],
                "target": "events",
              },
              "ts": Object {
                "ns": 0,
                "sec": 0,
              },
            },
          },
        },
      ],
      "data": Object {
        "filter": "a",
        "finding": Object {
          "search": Object {
            "pins": Array [
              "a",
              "b",
            ],
            "program": "c",
            "spaceId": "defaultId",
            "spaceName": "default",
            "spanArgs": Array [
              Object {
                "ns": 1000000,
                "sec": 0,
              },
              Object {
                "ns": 2000000,
                "sec": 0,
              },
            ],
            "target": "events",
          },
          "ts": Object {
            "ns": 0,
            "sec": 0,
          },
        },
      },
    },
    Object {
      "children": Array [],
      "data": Object {
        "filter": "e",
        "finding": Object {
          "search": Object {
            "pins": Array [],
            "program": "e",
            "spaceId": "defaultId",
            "spaceName": "default",
            "spanArgs": Array [
              Object {
                "ns": 1000000,
                "sec": 0,
              },
              Object {
                "ns": 2000000,
                "sec": 0,
              },
            ],
            "target": "events",
          },
          "ts": Object {
            "ns": 0,
            "sec": 0,
          },
        },
      },
    },
  ],
  "data": "ROOT",
}
`;
